src/transformers/pipelines/base.py

```diff
@@ -763,11 +763,15 @@ def __init__(
         if is_torch_available() and isinstance(device, torch.device):
             self.device = device
         else:
-            self.device = device if framework == "tf" else torch.device("cpu" if device < 0 else f"cuda:{device}")
+            self.device = (
+                device
+                if framework == "tf"
+                else torch.device(device if type(device) == str else "cpu" if device < 0 else f"cuda:{device}")
+            )
```

Comment:
Those are too many if and else for my little brain. Can we maybe do more if/elif/else ?